packages/fhir-location-management/src/components/ViewDetails/DetailsTabs/Inventory.tsx

```diff
@@ -0,0 +1,291 @@
+import React from 'react';
+import { useMls } from '../../../mls';
+import {
+  TableLayout,
+  useTabularViewWithLocalSearch,
+  SearchForm,
+  Column,
+} from '@opensrp/react-utils';
+import { Alert, Button, Col, Divider, Row } from 'antd';
+import { IGroup } from '@smile-cdr/fhirts/dist/FHIR-R4/interfaces/IGroup';
+import { listResourceType } from '../../../constants';
+import { IBundle } from '@smile-cdr/fhirts/dist/FHIR-R4/interfaces/IBundle';
+import { RbacCheck } from '@opensrp/rbac';
+import { Link, useHistory } from 'react-router-dom';
+import { PlusOutlined } from '@ant-design/icons';
+import {
+  donorCharacteristicCoding,
+  getCharacteristicWithCoding,
+  groupResourceType,
+  inventoryGroupCoding,
+  poNumberIdentifierCoding,
+  productCoding,
+  sectionCharacteristicCoding,
+  quantityCharacteristicCoding,
+} from '@opensrp/fhir-helpers';
+import { hasCode } from '../utils';
+
+export interface InventoryViewProps {
+  fhirBaseUrl: string;
+  locationId: string;
+}
+
+/**
+ * Returns an object that can be easily consumed that represents an inventory resource
+ *
+ * @param group - inventory group resource to be parsed
+ */
+function parseInventoryGroup(group: IGroup) {
+  const { member, characteristic, identifier } = group;
+
+  const sanitizeDate = (dateString?: string) => {
+    if (!dateString) return;
+    const sampleDate = new Date(dateString);
+    if (isNaN(sampleDate.getTime())) return;
+    return sampleDate;
+  };
+
+  // invariant one member representing the product
+  const firstMember = member?.[0];
+  const productReference = firstMember?.entity.reference;
```

Comment:
Should we also add an optional chaining operator on `entity` here or  are we sure it will always have a value?